File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        String res="";
        
        for(int i=1; i!=num.length()-1; i++)
        {
            if(num.charAt(i-1)==num.charAt(i) && num.charAt(i)==num.charAt(i+1))
            {
                if(res.length()==0)
                {
                    res=num.substring(i-1, i+2);
                }
                else if(num.charAt(i)>res.charAt(0))
                {
                    res=num.substring(i-1, i+2);
                }
            }
        }
        return  res;
    }
}

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        char res=0;
        
        for(int i=2; i<num.size(); i++)
        {
            if(num[i]==num[i-1] && num[i]==num[i-2])
            {
                res=max(res, num[i]);
            }
        }
        if(res==0)
        {
            return "";
        }
        else
        {
            return string(3,res);    
        }
    }
};